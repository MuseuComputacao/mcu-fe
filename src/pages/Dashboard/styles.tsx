import styled from 'styled-components/native';
import { style } from '../../globalStyles';

import { widthPercentageToDP as wp, heightPercentageToDP as hp } from "react-native-responsive-screen";

export const DashboardView = styled.View`
    display: flex;
    flex-direction: column;
    width: ${props => props.isOpen ? 'calc(100% - 250px)' : 'calc(100% - 80px)'};
`;

export const SignInTitleView = styled.View`
    display: flex;
    flexDirection: column;
    alignItems: center;
    margin-bottom: 25px;
`;