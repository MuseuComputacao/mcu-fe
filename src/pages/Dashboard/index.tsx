import React, { useEffect, useState } from 'react';
import Sidebar from '../../components/Sidebar/index'
import { View, Text } from 'react-native';
import { DashboardView } from './styles';

import AsyncStorage from '@react-native-async-storage/async-storage';

const Dashboard = () => {
  useEffect(() => {
    handleUserData()
  },[])

  const [isOpen, setIsOpen] = useState<boolean>(true);

  function getIsOpenProp(getIsOpen: boolean){
    setIsOpen(getIsOpen);
  }

  const [user, setUser] = useState<any>({})

  const handleUserData = async () => {
    const userData =  await AsyncStorage.getItem('@user')
    setUser(JSON.parse(userData))
  }

    return(
      <View style={{ display: 'flex', flexDirection: 'row', height: '100%' }}>
        <Sidebar func={getIsOpenProp}/>
        <DashboardView isOpen={isOpen}>
         <Text style={{fontSize: '23px', textAlign: 'left', margin: '50px', lineHeight: '1.6em'}}>
          Ol√°, <br/>
          <strong>{user.name}</strong> bem vindo ao admin do museu da computa√ß√£o da UFRJ. <br/>
          ‚ù§ ü§ñ ‚ù§
         </Text>
        </DashboardView>
      </View>
    )
}

export default Dashboard;