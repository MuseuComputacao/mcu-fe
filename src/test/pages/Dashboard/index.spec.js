import React from 'react';
import renderer from 'react-test-renderer';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import 'jest-styled-components'
import Dashboard from '../../../pages/Dashboard/index';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

describe('<Dashboard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});