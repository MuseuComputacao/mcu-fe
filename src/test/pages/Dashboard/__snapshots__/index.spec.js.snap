// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
    }
  }
>
  <View
    isOpen={true}
    style={
      Array [
        Object {
          "backgroundColor": "#18121E",
          "height": "100%",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
          "width": 250,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": "auto",
          "marginRight": 10,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <Text
        isOpen={true}
        style={
          Array [
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        Dashboard
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <Text
        isOpen={true}
        style={
          Array [
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        Gráficos
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <Text
        isOpen={true}
        style={
          Array [
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        Usuários
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        }
      }
    >
      <Text />
      <Text
        isOpen={true}
        style={
          Array [
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        Sair
      </Text>
    </View>
  </View>
  <View
    isOpen={true}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "calc(100% - 250px)",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 23,
          "textAlign": "center",
        }
      }
    >
      Olá 
       você é um 
      .
    </Text>
  </View>
</View>
`;
